### ISSUE TITLE
Can't share files anymore due to restrictive Android permission

### ISSUE THREAD

**Issue by scarf** [signalapp/Signal-Android#12548]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When I tap the + to share something and choose "File" my file chooser comes up, but after I select the file I want to share, nothing gets returned to Signal and I'm left with an empty message.  I noticed that the permissions for Signal "Files and Media" is set to "Allow access to media only" which restricts Signal to images/audio/video/etc only.  There should be another option "Allow management of all files" that would allow Signal to access all files.

### Steps to reproduce
- open "Note to self"
- tap +
- tap File
- pick a non-media file, such as .txt or .pdf etc.

**Actual result:** Signal doesn't attach the file to the message, instead nothing is attached
**Expected result:** Signal attaches the file to the message

### Device info

**Device:** Google Pixel 5a
**Android version:** 11
**Signal version:** 5.52.5

### Link to debug log

https://debuglogs.org/android/5.52.5/34a1ab023bbbc4d8270e77f23a302ae0d7d3c95fbce10473714df0ddc80c6d0c

**Comment by scarf** [comment]
Just shared the file I wanted with myself via Telegram Saved Messages and it worked fine.  I notice Telegram also only has "Allow access to media only" for "Files and media" permission, so guess there is more to it than that...

**Comment by deebash** [comment]
@scarf Unable to recreate the issue.

Please share screen recording if possible.

**Comment by scarf** [comment]
https://user-images.githubusercontent.com/2446600/198900836-4179b580-47c9-4123-a077-dd43abee563e.mp4

**Comment by greyson-signal** [comment]
What are you running on this Pixel 5a? All of your OS info comes back as 'unknown', and it looks like you don't have the default Files app?

**Comment by scarf** [comment]
Yeah it's rooted and I have XPrivacyLua installed, that can be the culprit in times like this.  but it's running stock Android 11 still.  I have the Files app but for some reason it isn't registered as an option when I try to attach a file (any idea what causes that?).  In XPrivacy, I've granted Signal all the permissions it wants though, except for Location.createFromParcel and TelephonyManager/getNetworkCountryIso and /getSimCountryIso, I don't think that would be causing this right?

Plus the same functionality is working with Telegram (and other apps too), and they have more restrictions activated than Signal does...
